packages/quantic/playwright/page-object/insightSetupObject.ts

```diff
@@ -0,0 +1,21 @@
+import {Locator, Page, expect} from '@playwright/test';
+
+export class InsightSetupObject {
+  constructor(private page: Page) {
+    this.page = page;
+  }
+
+  get initializationInput(): Locator {
+    return this.page.locator(
+      'c-quantic-insight-interface input[type="hidden"]'
+    );
+  }
+
+  async waitForInsightInterfaceInitialization(): Promise<void> {
+    await this.initializationInput.waitFor({state: 'attached'});
+    await expect(this.initializationInput).toHaveAttribute(
+      'is-initialized',
+      'true'
+    );
+  }
+}
```

Comment:
If possible this could be in a single file with the `searchObject` that would receive a `useCase` as a param in the constructor?

Would you prefer this? This is just a suggestion trying to group the code that is different depending on the useCase in the same file, so we can re-use the exact same object in all tests 
and the only thing that would change is the `useCase` we pass to the constructor?


```suggestion
export class QuanticInterfaceObject {
  private readonly initializationInput: Locator;

  constructor(private page: Page, private useCase: string) {
    this.page = page;
    this.initializationInput = this.page.locator(
      'c-quantic-insight-interface input[type="hidden"]'
    );
  }
  
  async waitForInterfaceInitialization: Promise<void> {
    if(this.useCase == 'insight') {
      await this.waitForInsightInterfaceInitialization();
    } else {
      await this.waitForSearchInterfaceInitialization(); // add this
    }
  }

  async waitForInsightInterfaceInitialization(): Promise<void> {
    await this.initializationInput.waitFor({state: 'attached'});
    await expect(this.initializationInput).toHaveAttribute(
      'is-initialized',
      'true'
    );
  }
}
```

Reply:
The goal I'm aiming for in this setup is to separate the concerns, and have Classes that are responsible of one concise and clear objective, to keep simplicity and clarity.

That's why for example I have  the class `InsightSetupObject` which takes care of a specific job of initializing an insight interface, this job is not needed for search so that's why I didn't mix it with search.

having QuanticInterfaceObject that groups handling the search and insight will motivate the presence of a code in a following shape:
```    
   if(this.useCase == 'insight') {
      // do this
    } else {
      // do this
    }
``` 
and for me this does not separate the concerns in an ideal way, especially that we have other use cases like case assist and recommendations so following this logic we would also support them in  `QuanticInterfaceObject` which I think will not scale really well.